server/mediaserver.go

```diff
@@ -41,36 +42,39 @@ var ErrRTMPPlay = errors.New("ErrRTMPPlay")
 var ErrRoundInit = errors.New("ErrRoundInit")
 var ErrStorage = errors.New("ErrStorage")
 var ErrDiscovery = errors.New("ErrDiscovery")
+var ErrUnknownStream = errors.New("ErrUnknownStream")
 
 const HLSWaitInterval = time.Second
 const HLSBufferCap = uint(43200) //12 hrs assuming 1s segment
 const HLSBufferWindow = uint(5)
+const StreamKeyBytes = 6
 
 const SegLen = 4 * time.Second
-const HLSUnsubWorkerFreq = time.Second * 5
 const BroadcastRetry = 15 * time.Second
 
-var HLSWaitTime = time.Second * 45
 var BroadcastPrice = big.NewInt(1)
 var BroadcastJobVideoProfiles = []ffmpeg.VideoProfile{ffmpeg.P240p30fps4x3, ffmpeg.P360p30fps16x9}
-var MinDepositSegmentCount = int64(75)     // 5 mins assuming 4s segments
-var MinJobBlocksRemaining = big.NewInt(40) // 10 mins assuming 15s blocks
+var MinDepositSegmentCount = int64(75) // 5 mins assuming 4s segments
 var LastHLSStreamID core.StreamID
 var LastManifestID core.ManifestID
 
+type rtmpConnection struct {
+	nonce  uint64
+	stream stream.RTMPVideoStream
+	pl     core.PlaylistManager
+	sess   *BroadcastSession
+}
+
 type LivepeerServer struct {
-	RTMPSegmenter   lpmscore.RTMPSegmenter
-	LPMS            *lpmscore.LPMS
-	LivepeerNode    *core.LivepeerNode
-	VideoNonce      map[string]uint64
-	VideoNonceLock  *sync.Mutex
-	HttpMux         *http.ServeMux
-	CurrentPlaylist core.PlaylistManager
+	RTMPSegmenter lpmscore.RTMPSegmenter
+	LPMS          *lpmscore.LPMS
+	LivepeerNode  *core.LivepeerNode
+	HttpMux       *http.ServeMux
 
 	ExposeCurrentManifest bool
 
-	rtmpStreams                map[core.StreamID]stream.RTMPVideoStream
-	broadcastRtmpToManifestMap map[string]string
+	rtmpConnections map[core.ManifestID]*rtmpConnection
```

Comment:
@j0sh I'm not sure I understand why `rtmpConnections` is part of the `LivepeerServer` struct which is shared by both B and O roles. What's the best way for me to better understand why Os need to hold variables only useful for B?

thanks!

Reply:
O doesn't use most of the fields in `LivepeerServer` -- in fact, the RTMP listener isn't even started for O.

What you're seeing is a reflection of how things have evolved up to this point, same as how we have a single `LivepeerNode` with fields that are only used by one or the other. We can clean this up at some point.